### PR TITLE
Softmax

### PR DESCRIPTION
**Pull request by pevnak**
This is a fix to an issue #68 (Softmax and crossentropy are prone to produce NaNs). I have added tests that the gradient does not produce any NaNs). I have also add more test of gradients of these functions to `runtests.jl` and also a check that gradients of these function do not overflows.

### CODE REVIEW

**Comment by MikeInnes** [comment]
Seems reasonable to me; we can't add Flux as a test dependency though.

**Comment by pevnak** [comment]
I have fixed that by adding calculated gradients.

**Comment by MikeInnes** [comment]
Works for me. Thanks!
